tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedWithTransactionOperation.cs

```diff
@@ -0,0 +1,115 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Bson;
+using MongoDB.Driver.Tests.Specifications.unified_test_format;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedWithTransactionOperation : IUnifiedWithTransactionOperation
+    {
+        private readonly BsonArray _operations;
+        private readonly TransactionOptions _options;
+        private readonly IClientSessionHandle _session;
+
+        public UnifiedWithTransactionOperation(
+            IClientSessionHandle session,
+            BsonArray operations,
+            TransactionOptions options)
+        {
+            _session = session;
+            _operations = operations;
+            _options = options;
+        }
+
+        public void Execute(Action<BsonDocument, bool, CancellationToken> assertOperationCallback, CancellationToken cancellationToken)
+        {
+            _session.WithTransaction(
+                callback: (session, token) =>
```

Comment:
Consider:

    callback: (session, _) =>

since the `token` parameter is not used.

Reply:
Oh. My bad, I intended to use `token` in `assertOperationCallback` call, but messed it up. I've updated the code to use the `token`.
Partially Done